packages/text-area/src/TextArea.tsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useMemo } from 'react';
+import { IdAllocator, Either, createDataProp } from '@leafygreen-ui/lib';
+import { css, cx } from '@leafygreen-ui/emotion';
+import { uiColors } from '@leafygreen-ui/palette';
+import { spacing, fontFamilies } from '@leafygreen-ui/tokens';
+
+const idAllocator = IdAllocator.create('textarea');
+const textAreaProp = createDataProp('area-selector');
+
+export const State = {
+  None: 'none',
+  Error: 'error',
+} as const;
+
+export type State = typeof State[keyof typeof State];
+
+const Mode = {
+  Light: 'light',
+  Dark: 'dark',
+} as const;
+
+type Mode = typeof Mode[keyof typeof Mode];
+
+const containerStyles = css`
+  display: flex;
+  flex-direction: column;
+`;
+
+const labelStyle = css`
+  font-size: 14px;
+  font-weight: bold;
+  line-height: 16px;
+  padding-bottom: 4px;
+`;
+
+const descriptionStyle = css`
+  font-size: 14px;
+  line-height: 16px;
+  font-weight: normal;
+  padding-bottom: 4px;
+  margin-top: 0px;
+  margin-bottom: 0px;
+`;
+
+const textAreaContainer = css`
+  margin-top: 4px;
+  position: relative;
+  display: flex;
+  align-items: center;
+  z-index: 0;
+`;
+
+const textAreaStyle = css`
+  width: 100%;
+  min-height: ${spacing[6]}px;
+  resize: none;
+  border-radius: 4px;
+  padding-left: 12px;
+  font-size: 14px;
+  font-weight: normal;
+  font-family: ${fontFamilies.default};
+  z-index: 1;
+
+  &:focus {
+    outline: none;
+  }
+
+  &:disabled {
+    cursor: not-allowed;
+  }
+`;
+
+const interactionRingStyles = css`
+  transition: all 150ms ease-in-out;
+  transform: scale(0.9, 0.8);
+  border-radius: 7px;
+  position: absolute;
+  top: -3px;
+  bottom: -3px;
+  left: -3px;
+  right: -3px;
+  pointer-events: none;
+  background-color: pink;
+
+  ${textAreaProp.selector}:hover ~ & {
```

Comment:
To clean this up a little, you can write these two styles as:

```
${textAreaProp.selector} {
  &:hover ~ &,
  &:focus ~ & {
    transform: scale(1);
  }
}
```

Reply:
👍

Reply:
This implementation actually doesn't work

Reply:
What does the CSS output look like? It should work.